subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/DefaultJavaInstallationRegistry.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.jvm.toolchain.internal;
+
+import org.gradle.api.GradleException;
+import org.gradle.api.JavaVersion;
+import org.gradle.api.file.FileCollection;
+import org.gradle.api.internal.file.FileCollectionFactory;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.provider.ProviderFactory;
+import org.gradle.internal.exceptions.Contextual;
+import org.gradle.internal.jvm.Jvm;
+import org.gradle.jvm.toolchain.JavaDevelopmentKit;
+import org.gradle.jvm.toolchain.JavaInstallation;
+import org.gradle.jvm.toolchain.JavaInstallationRegistry;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.util.Optional;
+
+public class DefaultJavaInstallationRegistry implements JavaInstallationRegistry {
+    private final JavaInstallationProbe installationProbe;
+    private final ProviderFactory providerFactory;
+    private final FileCollectionFactory fileCollectionFactory;
+
+    public DefaultJavaInstallationRegistry(JavaInstallationProbe installationProbe, ProviderFactory providerFactory, FileCollectionFactory fileCollectionFactory) {
+        this.installationProbe = installationProbe;
+        this.providerFactory = providerFactory;
+        this.fileCollectionFactory = fileCollectionFactory;
+    }
+
+    @Override
+    public JavaInstallation getInstallationForCurrentVirtualMachine() {
+        // TODO - should probably return a provider too
```

Comment:
❓given this would change the public api, shouldn't it be done right away, even if it's in a simple way?